### PR TITLE
Add NCP sample and NCP Host package v3.0.0

### PR DESCRIPTION
**Pull request by AdamZelikNS**

### CODE REVIEW

File: subsys/osif/zb_nrf_platform.c [hunk 1]
```diff
@@ -357,6 +370,14 @@ static void zboss_thread(void *arg1, void *arg2, void *arg3)
 	while (1) {
 		zboss_main_loop_iteration();
 	}
+
+#else
+	/* CONFIG_ZIGBEE_LIBRARY_NCP_DEV */
```
Comment (edmont) [inline subsys/osif/zb_nrf_platform.c:375]: Move to previous line together with `#else`?

File: subsys/osif/zb_nrf_platform.c [hunk 2]
```diff
@@ -342,8 +343,20 @@ int zigbee_init(void)
 	return 0;
 }
 
+#if IS_ENABLED(CONFIG_ZIGBEE_LIBRARY_NCP_DEV)
+/* Callout function called in zboss_app_main() right after starting ZBOSS. */
```
Comment (edmont) [inline subsys/osif/zb_nrf_platform.c:347]: We don't need docs here since it's already in the header file.

File: samples/ncp/README.rst [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+You can then use this sample together with the :ref:`Zigbee Light bulb <zigbee_light_bulb_sample>` to set up a basic Zigbee network.
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+The nRF5340 DK (``nrf5340dk/nrf5340/cpuapp``) is supported only for development.
```
Comment (edmont) [inline samples/ncp/README.rst:24]: Remove.

File: samples/ncp/README.rst [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+
+You can use any of the development kits listed above.
+
+.. note::
+    The nRF52840 Dongle uses a :ref:`different bootloader <zigbee_ncp_bootloader>` than other development kits.
```
Comment (edmont) [inline samples/ncp/README.rst:29]: Remove.

File: samples/ncp/README.rst [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+       ncs,zigbee-uart = &uart0;
+   };
+
+By default, Zephyr's logger uses ``uart0`` and the NCP sample communicates through the UART serialization using ``uart1``.
+The DTS overlay file configures ``uart1`` to be connected to the on-board J-Link instead of ``uart0``.
+As the result, Zephyr's logger ``uart0`` is available only through GPIO pins (**P1.00** and **P1.01**).
+
+The ``uart0`` pins are configured by devicetree overlay files for each supported development kit in the :file:`boards` directory.
+
+Communication through USB
```
Comment (edmont) [inline samples/ncp/README.rst:91]: Remove.

File: samples/ncp/README.rst [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+  To generate a DFU package, see `Generating DFU packages`_ in the nRF Util user guide.
+  Upgrading the dongle using this method requires putting the dongle into the DFU mode.
+  When in the DFU mode, you can use `nRF Util`_ for sending the upgrade image.
+  See `DFU over a serial USB connection`_ in the nRF Util user guide.
+
+  .. note::
+      By default, you can enter the DFU mode on the dongle using the pin reset.
+      Alternatively, you can call :c:func:`ncp_host_ota_run_bootloader` to trigger the bootloader on the dongle from the NCP Host application.
+
+FEM support
```
Comment (edmont) [inline samples/ncp/README.rst:227]: Remove.

**Comment by AdamZelikNS** [comment]
To do:

- [ ] Update libncp-dev.a 
- [ ] Update samples/ncp/README.rst to not mention nRF52, USB, ..
- [ ] Update (optionally?) ncp_host_v3.0.0 adding README, release_notes.rst

**Comment by milewr** [comment]
> To do:
> 
> * [ ]  Update libncp-dev.a
> * [ ]  Update samples/ncp/README.rst to not mention nRF52, USB, ..
> * [ ]  Update (optionally?) ncp_host_v3.0.0 adding README, release_notes.rst

README.rst file will be updated by another PR. You can even remove it from this PR.

**Review by edmont — APPROVED** [review]

File: subsys/osif/zb_nrf_platform.c (hunk 1 above)
Comment (AdamZelikNS) [reply subsys/osif/zb_nrf_platform.c]: fixed

File: samples/ncp/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
```
Comment (edmont) [inline samples/ncp/CMakeLists.txt:2]: For coherence with the other samples, we can update the copyright dates to 2024 in NCP sample files.

File: samples/ncp/prj.conf [hunk 8]
```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
```
Comment (edmont) [inline samples/ncp/prj.conf:2]: Ditto.

File: samples/ncp/src/main.c [hunk 9]
```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
```
Comment (edmont) [inline samples/ncp/src/main.c:2]: Ditto.

**Review by edmont — APPROVED** [review]

**Pull request merged**